llvm/lib/Transforms/Instrumentation/AddressSanitizer.cpp

```diff
@@ -2922,6 +2995,9 @@ bool ModuleAddressSanitizer::instrumentModule(Module &M) {
     auto *MD = M.getOrInsertNamedMetadata("device.sanitizer");
     Metadata *MDVals[] = {MDString::get(Ctx, "asan")};
     MD->addOperand(MDNode::get(Ctx, MDVals));
+
+    IRBuilder<> IRB(*C);
+    instrumentDeviceGlobal(IRB, M);
```

Comment:
how about moving IRB into instrumentDeviceGlobal?

Reply:
not important, and it's ok since its prototype is similar to other instrument functions